src/libraries/System.Private.CoreLib/src/System/Globalization/CompareInfo.Windows.cs

```diff
@@ -211,11 +200,22 @@ private unsafe int CompareString(ReadOnlySpan<char> string1, string string2, Com
 
             string? localeName = _sortHandle != IntPtr.Zero ? null : _sortName;
 
+            // CompareStringEx may try to dereference the first character of its input, even if an explicit
+            // length of 0 is specified. To work around potential AVs we'll always ensure zero-length inputs
```

Comment:
Really? What is the input that it happens for?

Reply:
Create a new C++ application and drop in this code. It will AV the process.

```cpp
	auto result = CompareStringEx(
		LOCALE_NAME_INVARIANT,
		NORM_LINGUISTIC_CASING,
		L"x",
		1,
		(LPCWCH)0xdeadbeef,
		0,
		nullptr,
		nullptr,
		0);
```